wgengine/router/router.go

```diff
@@ -72,6 +72,49 @@ type Config struct {
 	NetfilterMode    preftype.NetfilterMode // how much to manage netfilter rules
 }
 
+func (a *Config) Equal(b *Config) bool {
```

Comment:
it's probably adding a test like tailcfg's TestHostinfoEqual here, so a test fails if somebody adds a new field to Config and forgets to update this method.

At least just this part of the test:

```go
func TestHostinfoEqual(t *testing.T) {
	hiHandles := []string{
		"IPNVersion", "FrontendLogID", "BackendLogID",
		"OS", "OSVersion", "Desktop", "Package", "DeviceModel", "Hostname",
		"ShieldsUp", "ShareeNode",
		"GoArch",
		"RoutableIPs", "RequestTags",
		"Services", "NetInfo", "SSH_HostKeys",
	}
	if have := fieldsOf(reflect.TypeOf(Hostinfo{})); !reflect.DeepEqual(have, hiHandles) {
		t.Errorf("Hostinfo.Equal check might be out of sync\nfields: %q\nhandled: %q\n",
			have, hiHandles)
	}
```

Reply:
Would it also make sense to use `reflect.DeepEqual` in the implementation of `Equal` for `Config` (that's what `Hostinfo` does)?

Reply:
it's not as fast and can allocate and nil vs empty slices are different. So if those don't matter then that's fine.

Reply:
This is not a hotspot as far as I can tell, so I switched to it. Added tests too.